### DISCUSSION TITLE
How do people manage services on FreeBSD?

### DISCUSSION THREAD

**Discussion by vain** [bundlewrap/bundlewrap#638] [General]
I *think* this was once covered by `svc_systemv`, but I don’t run FreeBSD, so I don’t really know. These days, `/etc/init.d` does not exist anymore:

```
root@freebsd:~ # ls -al /etc/init.d
ls: /etc/init.d: No such file or directory
```

I think @ThreeFx is running FreeBSD? How do you manage services there?

I kind of wonder, do we still support FreeBSD properly? 🤔

(Also, I wanted to give these fancy GitHub “Discussions” a try, yay.)

**Comment by ThreeFx** [comment]
I haven't really gotten round to running proper stuff on FreeBSD yet, it's more a test range for me to play around with a different kernel/OS. I think that service management on FreeBSD is done using a combination of `sysrc` and `service`. I'm not sure whether I have started to write some code for that somewhere, but I can't find it at the moment...

**Comment by trehn** [comment]
I don't feel like we can commit to "officially support" anything maybe save for the very narrow Ubuntu environment we use ourselves. We should remain flexible enough to work on BSDs, but I have never put much deliberate effort in it. User demand just hasn't been there so far.

**Comment by simonhoffmann** [comment]
> I *think* this was once covered by `svc_systemv`, but I don’t run FreeBSD, so I don’t really know. These days, `/etc/init.d` does not exist anymore:
> 
> 
> 
> ```
> 
> root@freebsd:~ # ls -al /etc/init.d
> 
> ls: /etc/init.d: No such file or directory
> 
> ```
> 
> 
> 
> I think @ThreeFx is running FreeBSD? How do you manage services there?
> 
> 
> 
> I kind of wonder, do we still support FreeBSD properly? 🤔
> 
> 
> 
> (Also, I wanted to give these fancy GitHub “Discussions” a try, yay.)

`/etc/rc.d` and `/usr/local/etc/rc.d`

There is also the `service` command

**Comment by simonhoffmann** [comment]
Please see https://github.com/bundlewrap/bundlewrap/pull/681

ok?

Then i will provide documentation
